apps/api/src/applicant/ienapplicant.util.service.ts

```diff
@@ -317,4 +386,17 @@ export class IENApplicantUtilService {
     }
     return job_title;
   }
+
+  async updateLatestStatusOnApplicant(mappedApplicantList: string[]): Promise<void> {
+    try {
+      // update applicant with latest status
+      const idsToUpdate = `'${mappedApplicantList.join("','")}'`;
+      const queryToUpdate = `UPDATE ien_applicants SET status_id = (SELECT status_id FROM ien_applicant_status_audit asa WHERE asa.applicant_id=ien_applicants.id ORDER BY asa.start_date DESC limit 1) WHERE ien_applicants.id IN (${idsToUpdate})`;
```

Comment:
If you see a query, I have added a subquery with order and limit value. I was not able to do it in querybuilder for FK relation field.
As I need to fetch status with latest date, it could be added by us or sync from and HMBC ATS.

I have tried below code... and some other possible combination that we can apply. But no success it update null value.

.createQueryBuilder()
        .update(IENApplicant)
        .set({
          status: () =>
            '(SELECT status_id FROM ien_applicant_status_audit sa WHERE sa.applicant_id=ien_applicants.id ORDER BY asa.start_date DESC limit 1)',
        })
        .where('id IN (:...id)', { id: mappedApplicantList })
        .execute();